demo/web/firebase_element/x_login.dart

```diff
@@ -0,0 +1,122 @@
+/*
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+@HtmlImport('x_login.html')
+library polymer_elements.demo.web.firebase_element.x_login;
+
+import 'dart:html' as dom;
+import 'dart:convert' show JSON;
+import 'package:web_components/web_components.dart' show HtmlImport;
+import 'package:polymer/polymer.dart';
+import 'package:polymer_elements/firebase_auth.dart';
+
+/// Silence analyzer [FirebaseAuth]
+@PolymerRegister('x-login')
+class XLogin extends PolymerElement {
+  XLogin.created() : super.created();
+
+  @property String provider = 'anonymous';
+  @property String message = '';
+  @property String email = '';
+  @property String password = '';
+  @property String newPassword = ''; //
+  @property Map user;
+  @property bool statusKnown = false;
+  @property String params; //
+
+  @reflectable
+  void login([_, __]) {
+    String params;
+    Map parameters;
+    try {
+      parameters = JSON.decode(params);
+    } catch (e) {
+      parameters = null;
+    }
+
+    if (provider == 'password') {
+      parameters = parameters ?? {};
+      parameters['email'] = email;
+      parameters['password'] = password;
+    }
+
+    //_firebaseLogin.login(parameters, null);
+    _firebaseLogin.jsElement.callMethod('login', [parameters, null]);
+  }
+
+  FirebaseAuth get _firebaseLogin => $['firebaseLogin'];
+
+  @reflectable
+  void logout([_, __]) => _firebaseLogin.logout();
+
+  @reflectable
+  errorHandler(dom.Event e, var detail) =>
```

Comment:
variables, parameters and functions without type annotations always look like broken/experimental code to me. Removed anyway :)